benchmark/coniql_performance_test.py

```diff
@@ -45,15 +46,32 @@
     default="performance_test_results.txt",
     help="File to output results to",
 )
+parser.add_argument(
+    "-l",
+    "--log",
+    action="store_true",
+    dest="log_progress",
+    help="Indicate if client should log progress to tmp file",
+)
 
 
 class GraphQLClient:
-    def __init__(self, endpoint, signal, ws_protocol):
+    def __init__(self, endpoint, signal, ws_protocol, log_progress):
         self.endpoint = endpoint
         self.signal = signal
         self.ws_protocol = ws_protocol
+        self.first_subscribe = True
+        self.log = log_progress
+
+        if self.log:
+            print("Logging subscription progress")
+            self.progress_file = open("/tmp/progress.txt", "w")
 
     async def subscribe(self, idid, query, handle, n_messages):
+        monitor_progress = False
```

Comment:
I think it is needed. We have 2 cases that we need to protect against to make sure that there is only a single instance that logs progress:
1. In a single client there will be multiple `subscribe` calls - one for each subscription to a PV. We only need one of the subscriptions to log progress so I have picked the first one. The first `subscribe` will have `self.first_subscribe` set to `True` and so will enter the case statement. We then need to immediately switch `self.first_subscribe` to `False` so that subsequent subscribes don't trigger the logging. That means that we need the `monitor_progress` variable to save the fact that we should log.
2. Our bash script has the option to create multiple clients but we only want a single client to be able to log progress. Therefore I introduced the `self.log` parameter, which is passed in as a variable to the Python script and sets it to `True` for the first client (indicating it should log progress) and `False` for all other instances.

Reply:
Ah right, I had failed to see it was the same `GraphQLClient` for every subscription. Yes, with the current setup that does then make sense.